packages/desktop/test/specs/ws.spec.ts

```diff
@@ -0,0 +1,238 @@
+import assert = require('assert');
+import { promises as fs } from 'fs';
+import environments from '../libs/environments';
+import environmentsSettings from '../libs/environments-settings';
+import http from '../libs/http';
+import navigation from '../libs/navigation';
+import { WsConnection, withTimeout } from '../libs/ws';
+
+describe('WebSockets', () => {
+  it('should able to start a server with websockets', async () => {
+    // open env
+    await environments.open('ws');
+    await environments.start();
+  });
+
+  describe('Http and WebSocket mixed', () => {
+    it('should work http routes as usual among websocket routes', async () => {
+      await http.assertCall({
+        method: 'GET',
+        path: '/test/http',
+        testedResponse: {
+          status: 200,
+          body: 'This is a http response'
+        }
+      });
+    });
+  });
+
+  describe('One-to-one websocket', () => {
+    it('Should be able to connect to echo websocket service', async () => {
+      const ws = new WsConnection(3000, '/test/ws/echo');
+      await ws.openForConversation();
+      ws.assertWebsocketIsOpened();
+
+      await ws.assertReply(
+        '{ "message": "Hello world!" }',
+        'ECHO: { "message": "Hello world!" }'
+      );
+      await ws.assertReply(
+        '{ "message": "Hello world 2222!" }',
+        'ECHO: { "message": "Hello world 2222!" }'
+      );
+
+      ws.close();
+      ws.assertWebsocketIsClosed();
+    });
+
+    it('should be able to connect multiple clients to echo socket', async () => {
+      const ws1 = new WsConnection(3000, '/test/ws/echo');
+      await ws1.openForConversation();
+      ws1.assertWebsocketIsOpened();
+
+      const ws2 = new WsConnection(3000, '/test/ws/echo');
+      await ws2.openForConversation();
+      await ws2.assertWebsocketIsOpened();
+
+      await ws1.assertReply(
+        '{ "message": "Hello world 1!" }',
+        'ECHO: { "message": "Hello world 1!" }'
+      );
+      await ws2.assertReply(
+        '{ "message": "Hello world 2!" }',
+        'ECHO: { "message": "Hello world 2!" }'
+      );
+
+      ws1.assertWebsocketIsOpened();
+      ws2.assertWebsocketIsOpened();
+
+      ws1.close();
+
+      ws2.assertWebsocketIsOpened();
+
+      await ws2.assertReply(
+        '{ "message": "Hello world 3!" }',
+        'ECHO: { "message": "Hello world 3!" }'
+      );
+
+      ws2.close();
+    });
+
+    it('should return correct response for each message', async () => {
```

Comment:
@255kb 
Isn't this a test case fulfilling what you mean by the other thread? Or, do you have any other different scenario?

Reply:
It looks like it. I will retest soon :)

Reply:
It's working as you said, great!